wa/framework/configuration/parsers.py

```diff
@@ -140,6 +140,9 @@ def _pop_sections(self, raw):
         sections = raw.pop("sections", [])
         if not isinstance(sections, list):
             raise ConfigError('Invalid entry "sections" - must be a list')
+        for section in sections:
+            if not isinstance(section, dict):
```

Comment:
A better check her would be `if not hasattr(section, 'items')`.

Reply:
Updated.